File: test/metagen/config.test.js
var config = {
	projectID : "metatest",

	headerOutput : "metadata/include",
	sourceOutput : "metadata/src",

	metaItemCallback : processCallback,

	cppNamespace : "meta_test",

	metaClassFunctionPrefix : "buildMetaClass_",
	metaClassCreatePrefix : "createMetaClass_",

	metaClassMainRegisterPrefix : "registerMain_",

	sourceFilePrefix : "meta_test_",
	mainSourceFile : "register_meta_test",

	autoRegisterToGlobal : true,

	metaNamespace : "metatest",
	sourceHeaderCode : "" +
		"#include \"../../../testmetatraits.h\"\n"
	,
	sourceHeaderReplacer : [ "!.*test/metagen/include!i", "../../include" ],
	metaHeaderPath : "../include/",

	classTraits : [
		{ pattern : ".*\\b[io]stream$", traits : { copyConstructorHidden : true }  },
		{ pattern : ".*\\bw?string$", traits : { hasTypeConvertConstructor : true }  },
	],

	predefinedTemplateInstances : [
		"TemplateClassA<int, char, 5>", "",
		"TemplateClassB<TemplateBase>", "",
	],
};

function processCallback(item, data)
{
	var name = item.getPrimaryName();

	if(item.isMethod()) {
		if(name == "GetMaxMotorForce") {
			data.skipBind = true;
		}
	}

	if (("" + item.getQualifiedName()) === "SimpleOverride::createHelperData") {
		item.setDiscardResultOwnership(true);
	}

	if(item.isClass()) {
		if(name == "SimpleOverride") {
			data.getWrapperConfig().setWrapClass(true);
		}
		if(name == "SimpleOverridableIncrementer") {
			data.getWrapperConfig().setWrapClass(true);
		}
	}
}

File: test/metagen/config.test.lua
cpgf.import("cpgf", "builtin.debug")

myProjectID = "metatest"

function cppInclude(inc)
	return " -I \"" .. inc .. "\""
end

function onFileCallback(sourceFile)
--	sourceFile.addInclude("../../../testmetatraits.h");
end

function onMainCallback(item, data)
	local name = nil;
	if item.isNamed() then
		name = item.getName()
	end

	if item.isClass() then
		if name == "SimpleOverride" then
			data.setWrapClass(true);
		end
	end
end

config = {
	projectID = myProjectID,
	sourceRootPath = "./",
	
	cppNamespace = "meta_test",

	targetFilePrefix = "meta_test_",
	
	maxItemCountPerFile = 0,
	
	headerIncludePrefix = "../include/",

	clangOptions = "-x c++"
			.. " -std=c++11"
			.. " -fcxx-exceptions"
			.. " -fms-compatibility -fms-extensions -fdelayed-template-parsing"

			.. cppInclude("D:/Program Files (x86)/Microsoft Visual Studio 11.0/VC/include")
			.. cppInclude("include")
			.. cppInclude("../../include")
	,
	
	files = {
		"include/*.h"
	},
	
	headerOutputPath = "metadata/include",
	sourceOutputPath = "metadata/src",
	
	metaNamespace = "metatest",
	mainRegisterFileName = "register_meta_test",
	
	templateInstantiations = {
		{ "TemplateClassA<int, char, 5>", "" },
		{ "TemplateClassB<TemplateBase>", "" },
	},
	
	force = false,
	
	stopOnCompileError = false,
	
	fileCallback = onFileCallback,
	mainCallback = onMainCallback,
	headerReplaceCallback = function(s) return s end,
}


